agent/internal/container/container_api_test.go

```diff
@@ -215,3 +218,134 @@ func TestContainer(t *testing.T) {
 		})
 	}
 }
+
+func TestContainerStatus(t *testing.T) {
+	log.SetLevel(log.TraceLevel)
+
+	t.Log("building client")
+	rawCl, err := dclient.NewClientWithOpts(dclient.WithAPIVersionNegotiation(), dclient.FromEnv)
+	require.NoError(t, err)
+	defer func() {
+		if cErr := rawCl.Close(); cErr != nil {
+			t.Logf("closing docker client: %s", cErr)
+		}
+	}()
+	cl := docker.NewClient(rawCl)
+
+	tests := []struct {
+		name                string
+		image               string
+		entrypoint          []string
+		signalAtState       cproto.State
+		signalAtEventAction string
+		signal              syscall.Signal
+		failure             *aproto.ContainerFailure
+	}{
+		{
+			name:                "killed after start and before run",
+			image:               "ubuntu",
+			entrypoint:          []string{"echo", "hello"},
+			signalAtState:       cproto.Starting,
+			signalAtEventAction: "create",
+			signal:              syscall.SIGKILL,
+			failure: &aproto.ContainerFailure{
+				FailureType: aproto.ContainerAborted,
+				ErrMsg:      "killed before run",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			id := cproto.NewID()
+
+			// setup event listener
+			eventChannel := make(chan struct{})
+			errChannel := make(chan error)
+			go func() {
+				t.Logf("creating %s event listener for %s container", tt.signalAtEventAction, id.String())
+				filterArgs := filters.NewArgs()
+				filterArgs.Add("type", "container")
+				filterArgs.Add("event", tt.signalAtEventAction)
+				filterArgs.Add("event.Actor.ID", id.String())
+
+				eventListener, errListener := rawCl.Events(context.Background(),
+					types.EventsOptions{Filters: filterArgs})
+				timeout := time.After(10 * time.Second)
+				for {
+					select {
+					case <-eventListener:
```

Comment:
Do we need this waiting on the ```eventListener``` and another ```eventChannel```?

Can we just create the ```eventListener``` on the main thread on listen for it where we are listening for ```eventChannel``` currently? and also waiting on errChannel / timeout in the final select too?

Reply:
That's absolutely fair, originally I hadn't applied filters to the `eventListener` so we were getting a lot of additional events coming in, so I had set it up in separate goroutine to handle that, but it's over-engineering now since it's only looking for the one event. I'll get this changed made 😄